From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spliterash <me@spliterash.ru>
Date: Wed, 21 Dec 2022 23:47:37 +0300
Subject: [PATCH] Kotlin MC Init


diff --git a/build.gradle.kts b/build.gradle.kts
index 864c724005098cdcd0bbb16366a35a6575a5ecba..b61b00a68af622121fe5e6f0d2897af79e6083b5 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -21,6 +21,9 @@ configurations.api {
 }
 
 dependencies {
+    // Minetopia start
+    api("ru.spliterash:kotlin-mc-core:1.0.1")
+    // Minetopia end
     // api dependencies are listed transitively to API consumers
     api("com.google.guava:guava:31.0.1-jre")
     api("com.google.code.gson:gson:2.8.9")
